File: wrappers/s2i/python/test/transformer-template-app/MyTransformer.py

class MyTransformer(object):
    """
    Router template. 
    """
    
    def __init__(self):
        """
        Add any initialization parameters. These will be passed at runtime from the graph definition parameters defined in your seldondeployment kubernetes resource manifest.
        """
        print("Initializing")


    def transform_input(self,features,feature_names):
        """
        transform input.

        Parameters
        ----------
        features : array-like
        feature_names : array of feature names (optional)
        """
        print("Running identity transform")
        return features

    def transform_output(self,features,feature_names):
        """
        transform output.

        Parameters
        ----------
        features : array-like
        feature_names : array of feature names (optional)
        """
        print("Running identity transform")
        return features
        

File: wrappers/s2i/nodejs/test/transformer-template-app/MyTransformer.js
let MyTransformer = function() {};

MyTransformer.prototype.init = function() {
  console.log("Initializing Transform ...");
};

MyTransformer.prototype.transform_input = function(X, names) {
  console.log("Identity Transform ...");
  return X;
};

MyTransformer.prototype.transform_output = function(X, names) {
  console.log("Identity Transform ...");
  return X;
};

module.exports = MyTransformer;
